scripts/build-windows-binary

```diff
@@ -30,7 +30,7 @@ VERSION_FLAGS="
     -X ${K3S_PKG}/pkg/version.Program=${PROG}
     -X ${K3S_PKG}/pkg/version.Version=${VERSION}
     -X ${RKE2_PKG}/pkg/images.DefaultRegistry=${REGISTRY}
-    -X ${RKE2_PKG}/pkg/images.DefaultEtcdImage=rancher/hardened-etcd:${ETCD_VERSION}-build20230802
+    -X ${RKE2_PKG}/pkg/images.DefaultEtcdImage=rancher/hardened-etcd:${ETCD_VERSION}-build20231009
```

Comment:
Also need to change this in https://github.com/rancher/rke2/blob/45c21222401f736f35c0ceff66db695c195b89a5/scripts/build-binary#L27